runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -390,17 +390,23 @@ public String toString() {
      *
      * <p>See also {@link PipelineExecutor#awaitCompletion()}.
      */
-    public State awaitCompletion() throws Throwable {
+    @Override
+    public State waitUntilFinish() {
       if (!state.isTerminal()) {
         try {
           executor.awaitCompletion();
           state = State.DONE;
-        } catch (InterruptedException e) {
-          Thread.currentThread().interrupt();
-          throw e;
-        } catch (Throwable t) {
-          state = State.FAILED;
-          throw t;
+        } catch (Throwable e) {
+          if (e instanceof InterruptedException) {
+            Thread.currentThread().interrupt();
+          }
+          if (e instanceof RuntimeException) {
```

Comment:
As per the `@deprecated` note use `throwIfUnchecked`:
https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Throwables.java#L232

Or you can do:

```
Throwables.propagateIfInstanceOf(RuntimeException.class, e);
Throwables.propagateIfInstanceOf(Error.class, e);
throw new RuntimeException(e);
```